pytorch_pfn_extras/nn/modules/lazy.py

```diff
@@ -81,32 +81,27 @@ def _lazy_load_hook(
             missing_keys, unexpected_keys, error_msgs):
         """load_state_dict pre-hook function for lazy buffers and parameters.
 
-        The purpose of this hook is to adjust the current state and/or
-        ``state_dict`` being loaded so that a module instance serialized in
-        both un/initialized state can be deserialized onto both un/initialized
-        module instance.
+        The purpose of this hook is to check the current state and/or
+        ``state_dict`` being loaded and ensure that both are states
+        are properly initialized.
 
         See comment in ``torch.nn.Module._register_load_state_dict_pre_hook``
         for the details of the hook specification.
         """
-        for name in self.lazy_buffer_names:
-            # Avoid shape mismatch error when loading an initialized buffer
-            # onto an uninitialized module instance.
-            self.register_buffer(name, state_dict[prefix + name])
-
```

Comment:
This needs to be reverted.

Reply:
if we do `register_buffer` we will overwrite the device, so we won't solve the bug, we need to work a way around it